src/c++/perf_analyzer/inference_profiler.h

```diff
@@ -234,6 +234,11 @@ class InferenceProfiler {
         current_value += step;
       } while (((current_value <= end) || (end == static_cast<T>(NO_LIMIT))) &&
                (meets_threshold));
+      // If there was only one concurrency we swept over and it did not meet the
+      // stability threshold, we should return an error.
+      if (current_value == (start + step) && meets_threshold == false) {
```

Comment:
Do we need to fix the same issue in line 230 and 239 of the same file? Maybe we can remove the log printing from the `Profile` function and make it part of the error message here to avoid confusion?

Reply:
We discussed offline and found a solution for this with a new `is_stable` variable to distinguish between not meeting threshold and not being stable 👍

Reply:
Discussed offline and created ticket to address other PA use cases to make sure they return non-zero exit code upon instability.